NewForce-Cohort-9/tabloid-full-stack-api-all-stars
# Ticket #40: Admins Can Approve Posts

## Description
Admins can now approve posts. This goes for old posts that do not already have pre-approval. [Trello Ticket #40](https://trello.com/c/xC5zgJkZ/40-40-admins-can-approve-posts)

## Type of change

## Testing Instructions
1. `git add` and `git commit` whatever you're currently working on
2. `git checkout main`
3. `git fetch --all`
4. `git checkout bp-adminpostapproval`
5. If you don't see the most recent version of my files, `git pull origin bp-adminpostapproval`
6. If you haven't created the Tabloid database, please do so.
7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-9/tabloid-full-stack-api-all-stars/blob/main/SQL/01_Tabloid_Create_DB.sql)
9. Click the execute with debugger solid green button in purple Visual Studio to launch the api
10. In your terminal, launch the program within the root folder with the command `npm start`

### Then:
* In the browser, please go to Login on the nav bar and log in with this email: `foo@bar.com` if you are not already
* Click on the dropdown menu for `Posts`
* Then, `New Post`
* Enter `TestingApproval` for all fields (choose any of the categories)
* Click `Add Post`
* After confirming the new post, click on `Admin Post Management` in the Nav Bar
* Your new post should appear at the top of the list with a green `Approve` button underneath it
* Click the button and watch your post disappear from the list
* After this, click on the dropdown menu again for `Posts` and click `All Posts`
* You can now see your approved post in the `All Posts` list 

## Checklist:
